### ISSUE TITLE
Update insert fails

### ISSUE THREAD

**Issue by dizzzz** [eXist-db/exist#1663]
### What is the problem

Original report by Alexander Henket:

I get a `NullPointerException` when I try to “update insert” the first node into an element that has attributes.

```
org.xmldb.api.base.XMLDBException
	at org.exist.xmldb.LocalXPathQueryService.execute(LocalXPathQueryService.java:195)
	at org.exist.xmldb.LocalXPathQueryService.lambda$execute$1(LocalXPathQueryService.java:159)
	at org.exist.xmldb.function.LocalXmldbFunction.apply(LocalXmldbFunction.java:46)
	at org.exist.xmldb.AbstractLocal.withDb(AbstractLocal.java:196)
	at org.exist.xmldb.LocalXPathQueryService.execute(LocalXPathQueryService.java:158)
	at org.exist.client.QueryDialog$QueryThread.run(QueryDialog.java:573)
Caused by: java.lang.NullPointerException
	at org.exist.dom.persistent.ElementImpl.appendChildren(ElementImpl.java:587)
	at org.exist.xquery.update.Insert.eval(Insert.java:150)
	at org.exist.xquery.LetExpr.eval(LetExpr.java:99)
	at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
	at org.exist.xquery.PathExpr.eval(PathExpr.java:276)
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
	at org.exist.xquery.XQuery.execute(XQuery.java:261)
	at org.exist.xquery.XQuery.execute(XQuery.java:185)
	at org.exist.xmldb.LocalXPathQueryService.execute(LocalXPathQueryService.java:192)
	... 5 more
Caused by: java.lang.NullPointerException
	at org.exist.dom.persistent.ElementImpl.appendChildren(ElementImpl.java:587)
	at org.exist.xquery.update.Insert.eval(Insert.java:150)
	at org.exist.xquery.LetExpr.eval(LetExpr.java:99)
	at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
	at org.exist.xquery.PathExpr.eval(PathExpr.java:276)
	at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
	at org.exist.xquery.XQuery.execute(XQuery.java:261)
	at org.exist.xquery.XQuery.execute(XQuery.java:185)
	at org.exist.xmldb.LocalXPathQueryService.execute(LocalXPathQueryService.java:192)
	at org.exist.xmldb.LocalXPathQueryService.lambda$execute$1(LocalXPathQueryService.java:159)
	at org.exist.xmldb.function.LocalXmldbFunction.apply(LocalXmldbFunction.java:46)
	at org.exist.xmldb.AbstractLocal.withDb(AbstractLocal.java:196)
	at org.exist.xmldb.LocalXPathQueryService.execute(LocalXPathQueryService.java:158)
	at org.exist.client.QueryDialog$QueryThread.run(QueryDialog.java:573)
```
### What did you expect

for the query below: no NPE, this XML file:

```xml
<root attr="1">
  <child/>
</root>
```

### Describe how to reproduce or add a test

To reproduce, run these lines of code:

```xquery
let $f  := xmldb:store('/db/apps', 'ttt.xml', <root attr="1"/>)
let $u  := update insert <child/> into doc($f)/root

return doc($f)
```

If you remove `attr=“1”` from the first node, the "update insert” statement will be successful.
If you add a `<!— comment —>` under the root node, the "update insert” statement will be successful.

### Context information
Please always add the following information
- eXist-db version : 3.6
- Java version (Java8u141)
- Operating system (MacOs)
- 64 bit
- Any custom changes in e.g. conf.xml : none

**Comment by dizzzz** [comment]
Added testcases

**Comment by median1st** [comment]
Hi, is there any patch available for this bug until 3.7.0 is released? Thank you.

**Comment by dizzzz** [comment]
At least the 'domts' test need to be executed. For this I need help from @adamretter ....

**Comment by adamretter** [comment]
@median1st 4.0.0 will be released shortly with this fix.

**Comment by ahenket** [comment]
Confirmed fixed in 4.0.0. Thanks!
